Kudu.Contracts/Functions/FunctionSecretsJsonOps.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+
+namespace Kudu.Core.Functions
+{
+    public class FunctionSecretsJsonOps : IKeyJsonOps<FunctionSecrets>
+    {
+        public int RequireKeyCount()
+        {
+            return 1;
+        }
+
+        // have the schema related info enclosed in this class
+        public string GenerateKeyUglyJson(Tuple<string, string>[] keyPair, string functionRt, out string unencryptedKey)
+        {
+            unencryptedKey = keyPair[0].Item1;
```

Comment:
Kudu api called to create master/function keys only work for the default format (ie we don't take any parameters), names are hard coded in the string